core/src/main/java/org/apache/iceberg/SchemaParser.java

```diff
@@ -58,9 +59,17 @@ private SchemaParser() {
   private static final String VALUE_REQUIRED = "value-required";
 
   static void toJson(Types.StructType struct, JsonGenerator generator) throws IOException {
+    toJson(struct, null, generator);
+  }
+
+  static void toJson(Types.StructType struct, Integer schemaId, JsonGenerator generator) throws IOException {
```

Comment:
Looks like the struct `toJson` methods aren't called outside of this file. What do you think about making them private?